### DISCUSSION TITLE
Has anyone tried to use ReactFlow with DnD/sortable libraries such as react-beautiful-dnd and dnd-kit?

### DISCUSSION THREAD

**Discussion by bfzli** [xyflow/xyflow#4354] [Q&A]
So, I have a different use case with ReactFlow. A node represents a page, and pages have components. Let's assume we have added one page and some components. I'm trying to make them sortable for easy management. I tried using popular packages like react-beautiful-dnd and dnd-kit to do this, but neither of them worked. Some had issues when used in a ReactFlow playground.

Because of the complicated way ReactFlow handles zoom, the sorting gets messed up when zooming in or out. The smoothness is broken, and it becomes barely usable. I was wondering if someone could shed some light on this.

**Comment by moklick** [comment]
Please provide some code via https://new.reactflow.dev or a repo

**Comment by SreedharAvvari** [comment]
Hey @bfzli, were you able to get this working using dnd-kit? I'm also running into issues when canvas is zoomed in/out. Any good approaches?
